src/components/ButtonElement.ts

```diff
@@ -0,0 +1,34 @@
+import { FC } from '..';
+import { Button as ButtonSpec } from '@slack/types';
+import { ElementProps, BlockElement } from './BlockElement';
+import { ContainerProps } from './ContainerProps';
+import { joinTextChildren } from './Text';
+
+export interface ButtonElementProps
+  extends ElementProps<'button'>,
```

Comment:
what does extending `ElementProps` actually accomplish here? am i crazy/missing something, or could this be removed and still work exactly the same

Reply:
yeah so... still learning TS but there's a concept called [string literal types](https://mariusschulz.com/blog/string-literal-types-in-typescript) ([more info](https://basarat.gitbooks.io/typescript/docs/types/literal-types.html)) with the aim of being able to do stuff [like discriminated unions](https://basarat.gitbooks.io/typescript/docs/types/discriminated-unions.html).

and was hoping to use `T` for filling in the payload. but yeah doesn't do much right now except for annotating